webapi/CopilotChat/Extensions/SemanticKernelExtensions.cs

```diff
@@ -63,6 +64,18 @@ public static IKernel RegisterCopilotChatSkills(this IKernel kernel, IServicePro
         return kernel;
     }
 
+    /// <summary>
+    /// Propegate exception from within semantic function
+    /// </summary>
+    public static void ThrowIfFailed(this SKContext context)
+    {
+        if (context.ErrorOccurred)
+        {
+            context.Logger.LogError(context.LastException, "{0}", context.LastException?.Message);
+            throw context.LastException!;
```

Comment:
I don't know that we want to do this. I'm not saying we don't. I'm just saying that I don't know. This changes the paradigm...

Reply:
Do we know how the kernel will handle the exceptions?

Reply:
It translates it into the state of the context in scope the function being called.  I'll find the change...give me a bit.

Reply:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs#L199

```
    [Obsolete("Throw exception from SKFunction implementation instead. The Kernel will set the failure properties. " +
        "This method will be removed in a future release.")]
    [EditorBrowsable(EditorBrowsableState.Never)]
    public SKContext Fail(string errorDescription, Exception? exception = null)
    {
        // Temporary workaround: if no exception is provided, create a new one.
        this.LastException = exception ?? new KernelException(KernelException.ErrorCodes.UnknownError, errorDescription);
        return this;
    }
```

Reply:
Locating InvokeAsync...(in progress)